### PR TITLE
(pre-review) Add AIP for private entry function for multisig creation

### PR DESCRIPTION
**Pull request by yeptos**

### CODE REVIEW

File: aips/aip-x.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+## High-level Overview
+
+The proposal introduces two new private entry functions in the `multisig_account` module:
+
+- `create_with_existing_account_call`: Creates a new multisig account on top of an existing account without revoking the original auth key.
+
+- `create_with_existing_account_and_revoke_auth_key_call`: Creates a new multisig account on top of an existing account and immediately rotates the original auth key to 0x0.
+
+These functions simplify the process of creating a multisig account from an existing account, regardless of its authentication key scheme. This is particularly useful for migrating from multi-key accounts to multisig accounts, which is currently not possible.
```
Comment (gregnazario) [inline aips/aip-x.md]: I would mention this mirrors the changes here: https://github.com/aptos-labs/aptos-core/commit/d771cec3c4638a9a5d50120765f921e47feea83f

File: aips/aip-x.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+- `test_create_multisig_account_on_top_of_existing_with_signer`: Tests creating a multisig account on top of an existing account using a signer
+- `test_create_multisig_account_on_top_of_existing_and_revoke_auth_key_with_signer`: Tests creating a multisig account and revoking the auth key
+
+These tests cover the basic functionality of the new functions.
+
+## Risks and Drawbacks
+
+The `create_with_existing_account_and_revoke_auth_key_call` function rotates the auth key to 0x0 and revokes capability offers, which irreversibly changes account control. Developers and users should be well-informed about this consequence.
+
```
Comment (gregnazario) [inline aips/aip-x.md]: I would mention as with all of these, the more other components (including resource accounts and objects), it may not be clear necessarily if the rotated key is the only thing that can authorize a signer at the address.

File: aips/aip-x.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+1. This change could lead to increased adoption of multisig accounts in the Aptos ecosystem.
+2. It may inspire further improvements in account migration and management tools.
+3. The simplified creation process could encourage the development of more complex governance structures built on multisig accounts.
+
+## Timeline
+
+### Suggested implementation timeline
+
+// TODO
+
```
Comment (gregnazario) [inline aips/aip-x.md]: I would give a week for this, since you have the entire implementation already.

File: aips/aip-x.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+### Suggested developer platform support timeline
+
+// TODO
+
```
Comment (gregnazario) [inline aips/aip-x.md]: This would probably need a week for all of the SDKs.  Otherwise, it may not even need to be a predetermined function, and then it can just be called like any other entry function.  Especially since there is proof created.

File: aips/aip-x.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+### Suggested deployment timeline
+
+// TODO
```
Comment (gregnazario) [inline aips/aip-x.md]: I would just mention in the next possible release, and any testing required.

File: aips/aip-x.md (hunk 2 above)
Comment (yeptos) [reply aips/aip-x.md]: could you please explain this in simpler terms? i'm not sure i understood this

File: aips/aip-x.md (hunk 4 above)
Comment (yeptos) [reply aips/aip-x.md]: yeah since it's a simple entry function like any other, I don't think we need any special support from the SDK.

File: aips/aip-x.md (hunk 2 above)
Comment (yeptos) [reply aips/aip-x.md]: i got what you meant after reading the implementation review comment. change applied
